src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java

```diff
@@ -133,36 +148,41 @@ public void firstPassApply(final VariantContext variant, final ReadsContext read
 
     @Override
     public void afterFirstPass() {
-        logger.info(String.format("Found %d SNPs %d indels with INFO score key:%s.", scoredSnps, scoredIndels, infoKey));
-        logger.info(String.format("Found %d SNPs %d indels in the resources.", snpScores.size(), indelScores.size()));
+        logger.info(String.format("Found %d SNPs and %d indels with INFO score key:%s.", scoredSnps, scoredIndels, infoKey));
+        logger.info(String.format("Found %d SNPs and %d indels in the resources.", resourceSNPScores.size(), resourceIndelScores.size()));
 
-        if (scoredSnps == 0 || scoredIndels == 0 || snpScores.size() == 0 || indelScores.size() == 0){
-            throw new GATKException("VCF must contain SNPs and indels with scores and resources must contain matching SNPs and indels.");
+        if (scoredSnps == 0 || scoredIndels == 0 || resourceSNPScores.size() == 0 || resourceIndelScores.size() == 0){
+            throw new UserException("VCF must contain SNPs and indels with scores and resources must contain matching SNPs and indels.");
         }
 
-        Collections.sort(snpScores, Collections.reverseOrder());
-        Collections.sort(indelScores, Collections.reverseOrder());
+        Collections.sort(resourceSNPScores, Collections.reverseOrder());
+        Collections.sort(resourceIndelScores, Collections.reverseOrder());
+
+        for(double t : snpTranches) {
+            int snpIndex = (int)((t/100.0)*(double)(resourceSNPScores.size()-1));
+            snpCutoffs.add(resourceSNPScores.get(snpIndex));
+        }
 
-        for(double t : tranches) {
-            int snpIndex = (int)((t/100.0)*(double)(snpScores.size()-1));
-            snpCutoffs.add(snpScores.get(snpIndex));
-            int indelIndex = (int)((t/100.0)*(double)(indelScores.size()-1));
-            indelCutoffs.add(indelScores.get(indelIndex));
+        for(double t : indelTranches) {
+            int indelIndex = (int)((t/100.0)*(double)(resourceIndelScores.size()-1));
+            indelCutoffs.add(resourceIndelScores.get(indelIndex));
         }
 
     }
 
     @Override
     protected void secondPassApply(VariantContext variant, ReadsContext readsContext, ReferenceContext referenceContext, FeatureContext featureContext) {
         final VariantContextBuilder builder = new VariantContextBuilder(variant);
-
+        if (removeOldFilters) {
+            builder.unfiltered();
+        }
         if (variant.hasAttribute(infoKey)) {
             final double score = Double.parseDouble((String) variant.getAttribute(infoKey));
             if (variant.isSNP() && isTrancheFiltered(score, snpCutoffs)) {
-                builder.filter(filterStringFromScore(score, snpCutoffs));
+                builder.filter(filterStringFromScore("SNP", score, snpTranches, snpCutoffs));
```

Comment:
Extract string constants for "SNP" and "INDEL" and use them in the calls to `filterStringFromScore ` and `addTrancheHeaderFields'.

Reply:
done